liblangutil/Scanner.cpp

```diff
@@ -271,12 +273,74 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+// A little helper class for counting depth of RLO/LRO/PDF uses.
+struct Scanner::UnicodeDirectionOverrideProcessor
+{
+	Scanner& scanner;
+	int rtlOverrideDepth = 0;
+
+	/// Tries to scan for an RLO/LRO/PDF and keeps track of script writing direction override depth.
+	///
+	/// @returns a 2-tuple indicating whether or not successfully some script writing direction
+	///          override has been consumed from the input and an error code in case
```

Comment:
```suggestion
	/// @returns a 2-tuple indicating whether or not some script writing direction
	///          override has been consumed successfully from the input and an error code in case
```